src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_handler.js

```diff
@@ -57,9 +57,14 @@ export class VisualizeEmbeddableHandler extends EmbeddableHandler {
           visualizeScope.$destroy();
         });
 
+        const renderComplete = new Promise(resolve => {
+          visualizeScope.$on('ready:vis', resolve);
```

Comment:
this event is emitted far before visualization is actually done rendering (after linking of angular directive is done)

there is also a renderComplete event emitted by visualize, when visualization is actually done rendering, however that is a native event emitted on the <visualize> dom node.

Reply:
hmm, maybe I misunderstood the meaning of the `ready:vis` signal.  If it doesn't mean "i'm done rendering", what does it indicate?  I think this is just a naming issue, right? 

I wonder if the timing is important to dashboard (it's old code so I'm not entirely sure).  Why listen to this instead of the `renderComplete`. 

I could rename the variable `readyVis` just to get rid of the broadcasting, but it's pretty unclear what that even means, and how and when other embeddables should be resolving it.

Reply:
`ready:vis` just tells the linking is done .... it was like this before, so i guess dashboard doesn't really care when rendering is actually done, but just wanted to bring this up (to raise awareness ;) )

Reply:
hmm, maybe the important part is that it's telling dashboard that it's request has been added to courier?  Since dashboard uses this signal to tell courier to fetch it, and that's how the msearch batches work?   I suppose I could rename requestsQueued.

I've been thinking about how this should look long term, but am not sure yet.  Should embeddables send their requests through dashboard so dashboard can be intelligent about batching them up? Or should we continue to have embeddables speak directly to courier to send in their requests.  I'm thinking the former is the better path forward, but we still can't stop embeddables from making direct requests.

But for the time being, just to get rid of the broadcast signal, I'll rename to requestsQueued and add some comments.